src/app_service/service/ens/service.nim

```diff
@@ -0,0 +1,486 @@
+
+      let contractDto = self.getCurrentNetworkContractForName("ens-usernames")
+      let sntContract = self.getCurrentNetworkErc20ContractForSymbol()
+
+      let price = getPrice(contractDto)
+      let label = fromHex(FixedBytes[32], label(ensUsername))
+
+      let register = Register(label: label, account: parseAddress(address), x: x, y: y)
+      let registerAbiEncoded = contractDto.methods["register"].encodeAbi(register)
+      let approveAndCallObj = ApproveAndCall[132](to: contractDto.address, value: price, 
+      data: DynamicBytes[132].fromHex(registerAbiEncoded))
+      let approveAndCallAbiEncoded = sntContract.methods["approveAndCall"].encodeAbi(approveAndCallObj)
+
+      var tx = buildTokenTransaction(parseAddress(address), sntContract.address, "", "")
+      var success = false
+
+      let response = sntContract.methods["approveAndCall"].estimateGas(tx, approveAndCallObj, success)
+
+      if(success):
+        result = fromHex[int](response)
+      else:
+        result = 380000
+
+    except RpcException as e:
+      error "error occurred", methodName="registerENSGasEstimate"
+
+  proc registerEns*(self: Service, username: string, address: string, gas: string, gasPrice: string, 
+    maxPriorityFeePerGas: string, maxFeePerGas: string, password: string): string =
+    try:
+      let myPublicKey = self.settingsService.getPublicKey()
+      let coordinates = self.extractCoordinates(myPublicKey)
+      let x = fromHex(FixedBytes[32], coordinates.x)
+      let y =  fromHex(FixedBytes[32], coordinates.y)
+
+      let contractDto = self.getCurrentNetworkContractForName("ens-usernames")
+      let sntContract = self.getCurrentNetworkErc20ContractForSymbol()
+
+      let price = getPrice(contractDto)
+      let label = fromHex(FixedBytes[32], label(username))
+      let eip1559Enabled = self.settingsService.isEIP1559Enabled()
+
+      let register = Register(label: label, account: parseAddress(address), x: x, y: y)
+      let registerAbiEncoded = contractDto.methods["register"].encodeAbi(register)
+      let approveAndCallObj = ApproveAndCall[132](to: contractDto.address, value: price, 
+      data: DynamicBytes[132].fromHex(registerAbiEncoded))
+
+      var tx = buildTokenTransaction(parseAddress(address), sntContract.address, gas, gasPrice, eip1559Enabled, 
+      maxPriorityFeePerGas, maxFeePerGas)
+      var success = false
+
+      let response = sntContract.methods["approveAndCall"].send(tx, approveAndCallObj, password, success)
+      result = $(%* { "result": %response, "success": %success })
+      if success:
+        var ensUsername = self.formatUsername(username, true)
+        self.transactionService.trackPendingTransaction(response, address, $sntContract.address, 
+        $PendingTransactionTypeDto.RegisterENS, ensUsername)
+        self.pendingEnsUsernames.incl(ensUsername)
+    
+    except RpcException as e:
+      error "error occurred", methodName="registerEns"
+
+  proc getSNTBalance*(self: Service): string =
```

Comment:
This function can probably be moved to the wallet or balance service, but it's not a big deal

Reply:
@jrainville yes, I mentioned that in the first comment [here](https://github.com/status-im/status-desktop/pull/4490#issue-1107855700), actually when I started to work on this I expected that many of the things I refactored here were already done within wallet section (and related things) refactoring done, but seems not. For now we will keep this, and when one is going to refactor other part which will need some method from `ens` service he will move it to the final, proper place.